crates/relayer-cli/src/commands/keys/add.rs

```diff
@@ -187,44 +195,63 @@ pub fn add_key(
     config: &ChainConfig,
     key_name: &str,
     file: &Path,
-    hd_path: &HDPath,
+    hd_path: &StandardHDPath,
     overwrite: bool,
-) -> eyre::Result<KeyEntry> {
-    let mut keyring = KeyRing::new(Store::Test, &config.account_prefix, &config.id)?;
+) -> eyre::Result<AnySigningKeyPair> {
+    let key_pair = match config.r#type {
+        ChainType::CosmosSdk => {
+            let mut keyring = KeyRing::new(Store::Test, &config.account_prefix, &config.id)?;
 
-    check_key_exists(&keyring, key_name, overwrite);
+            check_key_exists(&keyring, key_name, overwrite);
 
-    let key_contents = fs::read_to_string(file).map_err(|_| eyre!("error reading the key file"))?;
-    let key = KeyEntry::from_seed_file(&key_contents, hd_path)?;
+            let key_contents =
+                fs::read_to_string(file).map_err(|_| eyre!("error reading the key file"))?;
+            let key_pair = Secp256k1KeyPair::from_seed_file(&key_contents, hd_path)?;
 
-    keyring.add_key(key_name, key.clone())?;
-    Ok(key)
+            keyring.add_key(key_name, key_pair.clone())?;
+            key_pair.into()
+        }
+    };
+
+    Ok(key_pair)
 }
 
 pub fn restore_key(
     mnemonic: &Path,
     key_name: &str,
-    hdpath: &HDPath,
+    hdpath: &StandardHDPath,
     config: &ChainConfig,
     overwrite: bool,
-) -> eyre::Result<KeyEntry> {
+) -> eyre::Result<AnySigningKeyPair> {
     let mnemonic_content =
         fs::read_to_string(mnemonic).map_err(|_| eyre!("error reading the mnemonic file"))?;
 
-    let mut keyring = KeyRing::new(Store::Test, &config.account_prefix, &config.id)?;
+    let key_pair = match config.r#type {
+        ChainType::CosmosSdk => {
+            let mut keyring =
+                KeyRing::<Secp256k1KeyPair>::new(Store::Test, &config.account_prefix, &config.id)?;
```

Comment:
Let's introduce a constructor here rather have to specify the type? eg.

```suggestion
                KeyRing::new_secp256k1(Store::Test, &config.account_prefix, &config.id)?;
```

Reply:
And one for the ed25519 keyring

Reply:
Added these functions.